authLdap.php

```diff
@@ -310,6 +326,48 @@ function authLdap_login($user, $username, $password, $already_md5 = false)
 		$loggedInUser = $mapper($loggedInUser);
 	}
 
+	// Set global variable to distinguish LDAP and non-LDAP logins for authorization flow
+	global $authLDAPisLdapLogin;
+	$authLDAPisLdapLogin = true;
```

Comment:
Do we really need this? After all the process right now consists of 2 completely separate checks: Authenticate and Authorize. Why does the Authorize need to know where the authentication was done?

Reply:
Yes, but it's a bit hard to articulate why. Basically, we have 3 potential sources of user authentication (generally speaking):
- LDAP users
- Internal WP users
- "Other"/External users

Before we invoke Authorize, we need to know if we can recycle the existing LDAP connection - which will only work if we successfully authenticated via LDAP in the authLdap_login function. If we blindly reset the LDAP connection regardless of the authentication method (which we _could_ do, of course), it overwrites the functionality of the UserRead flag.

So this global is there to ensure that a successful LDAP authentication continues to use the existing LDAP connection, and in all other cases, force it to reset because without that reset, the LDAP connection doesn't function.

Reply:
Hm. Let me check why I introduced that flag. There was a reason. I'll get back to you on that later